app/models/manageiq/providers/openstack/inventory/collector/cloud_manager.rb

```diff
@@ -103,33 +115,47 @@ def quotas
   end
 
   def servers
-    @servers ||= compute_service.handled_list(:servers)
+    @servers ||= compute_service.handled_list(:servers, {}, ::Settings.ems.ems_openstack.refresh.is_admin)
   end
 
   def servers_by_id
     @servers_by_id ||= Hash[servers.collect { |s| [s.id, s] }]
   end
 
   def tenants
-    @tenants ||= manager.openstack_handle.accessible_tenants
+    if ::Settings.ems.ems_openstack.refresh.is_admin
+      @tenants ||= identity_service.visible_tenants.select do |t|
+        # avoid 401 Unauth errors when checking for accessible tenants
+        # the "services" tenant is a special tenant in openstack reserved
+        # specifically for the various services
+        next if t.name == "services"
+        true
+      end
+    else
+      @tenants ||= manager.openstack_handle.accessible_tenants
+    end
```

Comment:
Is there a clean way to put the weird 'services' filter into one place code-wise, or is that not really a practical option?